### PR TITLE
Fix copy paste errors, add warning to readme

### PR DESCRIPTION
**Pull request by SgtCoDFish**
See individual commit messages for more context!

Also updates the boilerplate checker.

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module github.com/cert-manager/csi-driver/hack/tools
```
Comment (JoshVanL) [inline hack/tools/go.mod:1]:
```suggestion
module github.com/cert-manager/trust/hack/tools
```

**Review by JoshVanL — CHANGES_REQUESTED** [review]
One thing

/assign @SgtCoDFish

File: hack/tools/go.mod (hunk 1 above)
Comment (SgtCoDFish) [reply hack/tools/go.mod]: Ah good spot, cheers! Fixed now :grin:

**Pull request merged**